File: src/main/java/com/github/codingvampyre/drachenfest/domain/Account.java
package com.github.codingvampyre.drachenfest.domain;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.validation.constraints.NotNull;
import java.util.UUID;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
@Entity
@Builder
public class Account extends BaseEntity {

    @Column(unique = true)
    @NotNull
    private String email;

    @NotNull
    private String password;

    @Column(unique = true)
    @NotNull
    private UUID uuid;

    @Enumerated(EnumType.STRING)
    @NotNull
    private Role role = Role.Participant;

}

File: src/main/java/com/github/codingvampyre/drachenfest/domain/Account.kt
package com.github.codingvampyre.drachenfest.domain

import java.util.*
import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
open class Account(
		@Column(nullable = false)
		open var password: String? = null,

		@Column(unique = true, nullable = false)
		open var email: String? = null,

		@Column(unique = true)
		open var uuid: @NotNull UUID? = null,

		@Enumerated(EnumType.STRING)
		open var role: @NotNull Role? = Role.Participant,

		@OneToMany(mappedBy = "account")
		open var tickets: MutableList<Ticket> = mutableListOf(),

		@OneToMany(mappedBy = "account")
		open var purchases: MutableList<Purchase> = mutableListOf(),
): BaseEntity()
